api/client/stack/deploy.go

```diff
@@ -52,6 +52,14 @@ func runDeploy(dockerCli *client.DockerCli, opts deployOptions) error {
 		return err
 	}
 
+	info, err := dockerCli.Client().Info(context.Background())
+	if err != nil {
+		return err
+	}
+	if !info.Swarm.ControlAvailable {
+		return fmt.Errorf("This node is not a swarm manager. Use \"docker swarm init\" or \"docker swarm join\" to connect this node to swarm and try again.")
+	}
```

Comment:
Wonder if we could reuse something like https://github.com/docker/docker/blob/c388a5bdfa922d85373cc8a77a08612ca4b67861/daemon/cluster/cluster.go#L765-L775, but I guess that means moving the check to the daemon